tests/integration/integration_nodejs_test.go

```diff
@@ -489,6 +489,44 @@ func TestProviderSecretConfig(t *testing.T) {
 	})
 }
 
+func TestResourceWithSecretSerializationNodejs(t *testing.T) {
```

Comment:
We had this existing test in `tests/integration/secret_outputs` from https://github.com/pulumi/pulumi/pull/2875 but we somehow lost the actual Go code to run the test at some point. I added it back for Node.js, and augmented it with use of `additionalSecretOutputs`, and also added the same test for Python.